src/modules/hip/kernel/non_silent_region_detection.hpp

```diff
@@ -0,0 +1,419 @@
+            atomicMin(&beginResult, beginIdx);
+            if(beginResult != srcLength)
+                break;
+        }
+    }
+
+    // Find the end index in src whose value is >= cutOffMag
+    for (int i = id_x; i < srcLength; i += hipBlockDim_x)
+    {
+        uint srcIdx = stridePerSample + srcLength - 1 - i;
+        if (srcPtr[srcIdx] >= cutOffMag)
+        {
+            endIdx = srcLength - 1 - i;
+            atomicMax(&endResult, endIdx);
+            if(endResult != 0)
+                break;
+        }
+    }
+
+    // Final store to dst
+    if(hipThreadIdx_x == 0)
+    {
+        if(beginResult == srcLength || endResult == 0)
+        {
+            beginTensor[id_z] = 0;
+            lengthTensor[id_z] = 0;
+        }
+        else
+        {
+            int detectBegin = beginResult;
+            int detectEnd = endResult - beginResult + 1;
+
+            // if both starting index and length of nonsilent region is not 0
+            // adjust the values as per the windowLength
+            if(detectBegin != 0 && detectEnd != 0)
+            {
+                int newBegin = max(detectBegin - (windowLength - 1), 0);
+                detectEnd += detectBegin - newBegin;
+                detectBegin = newBegin;
+            }
+            beginTensor[id_z] = detectBegin;
+            lengthTensor[id_z] = detectEnd;
+        }
+    }
+}
+
+// -------------------- Set 4 -  host helpers for kernel executor --------------------
+
+// return the nearest previous power of 2 for the given number
+inline Rpp32s prev_pow2(Rpp32s n)
+{
+    Rpp32s pow2 = 1;
+    while (n - pow2 > pow2)
+        pow2 += pow2;
+
+    return pow2;
+}
+
+// return the nearest next power of 2 for the given number
+inline Rpp32s next_pow2(Rpp32s n)
+{
+    Rpp32s pow2 = 1;
+    while (n > pow2)
+        pow2 += pow2;
+
+    return pow2;
+}
+
+// -------------------- Set 5 - non silent region kernels executor --------------------
+
+RppStatus hip_exec_non_silent_region_detection_tensor(Rpp32f *srcPtr,
+                                                      RpptDescPtr srcDescPtr,
+                                                      Rpp32s *srcLengthTensor,
+                                                      Rpp32s *detectedIndexTensor,
+                                                      Rpp32s *detectionLengthTensor,
+                                                      Rpp32f cutOffDB,
+                                                      Rpp32s windowLength,
+                                                      Rpp32f referencePower,
+                                                      Rpp32s resetInterval,
+                                                      rpp::Handle& handle)
+{
+    // check if scratch memory size required for moving mean square is within the limits
+    if ((srcDescPtr->n * srcDescPtr->strides.nStride) > 76800000) // 76800000 is the maximum scratch memory size needed for MMS buffer in RNNT training
```

Comment:
please #define this 76800000 number and use it here.

Reply:
Changed to constant and CTest added here -> https://github.com/ROCm/rpp/pull/395/files#diff-df94f1927a53d83c734f3a0e7d81a532662ca9505bf1c1cc788597ed108489cdR125